Update network_increase_bandwidth.md document with details for u18.04+

JohnHammell opened the pull request:
The current document does not work properly for Ubuntu 22.04 and probably any version >= 18.04.

I updated this page to include details on how to do this for versions 18.04+ and kept the original section, renaming that to be for versions <= 17.04.

Also added an example command at the bottom on how to change the queue.tx.length in the default LXD profile so that change would apply to all containers.

ru-fu commented:
Thanks a lot @JohnHammell !

Can you please sign the CLA so we can get this change in? See here: https://ubuntu.com/legal/contributors

JohnHammell commented:
@ru-fu thanks for pointing that out and linking to it. I signed / submitted the CLA a few minutes ago.

JohnHammell commented:
@ru-fu Is there anything I need to do in order to re-run and/or correct the failing checks for this pull request? Or is it currently fine as-is? I'm a bit new to submitting pull requests to Canonical, so wanted to make sure I'm doing this correctly. Thanks.

ru-fu commented:
> @ru-fu Is there anything I need to do in order to re-run and/or correct the failing checks for this pull request? Or is it currently fine as-is? I'm a bit new to submitting pull requests to Canonical, so wanted to make sure I'm doing this correctly. Thanks.

Hi @JohnHammell - I re-ran the action, and it seems to be an issue with the check right now. If you signed the CLA with your GitHub user name (and the correct capitalization), you should be all set and the problem is on our side. I'll check. :)

JohnHammell commented:
@ru-fu Am I supposed to fix things with this pull request until all checks pass, before it can be reviewed?

If so, how do I trigger the checks to re-run after making corrections?

The one check failing (Commits / Branch target and CLA) states no launchpad account. However, I do have a launchpad account and did include that in the CLA form as it cannot be submitted without the launchpad account part. Does my email used on Github need to match my email address on Launchpad? The CLA did not require that they match, the form was able to submit with them being different, but the error message in the check seems to be looking for a Launchpad account that matches my email address used on Github rather than the Launchpad account that I specified on the CLA.

The other check that's failing (Tests / Documentation), in the details it mentions 2 lines with trailing spaces and another line fails with the message "Ordered list item prefix." Do I need to "Convert to draft" to make those corrections?

ru-fu commented on doc/howto/network_increase_bandwidth.md:
```diff
@@ -17,7 +17,30 @@ In general, you should use small `txqueuelen` values with slow devices with a hi
+Complete the following steps to increase the network bandwidth on the LXD host:
+
+1. Increase the transmit queue length (`txqueuelen`) of both the real NIC (for example, `enp5s0f1`) and the LXD NIC (for example, `lxdbr0`).
+   To make the change permanent, create file /etc/udev/rules.d/60-custom-txqueuelen.rules with the following:
```
```suggestion
   To make the change permanent, create a file name `/etc/udev/rules.d/60-custom-txqueuelen.rules` with the following content:
```

ru-fu commented on doc/howto/network_increase_bandwidth.md:
```diff
@@ -17,7 +17,30 @@ In general, you should use small `txqueuelen` values with slow devices with a hi
+       KERNEL=="enp5s0f1", RUN+="/sbin/ip link set %k txqueuelen 10000"
+       KERNEL=="lxdbr0", RUN+="/sbin/ip link set %k txqueuelen 10000"
+      
+   Apply the above udev rules via:
```
```suggestion
   Apply the above `udev` rules via:
```

ru-fu commented on doc/howto/network_increase_bandwidth.md:
```diff
@@ -17,7 +17,30 @@ In general, you should use small `txqueuelen` values with slow devices with a hi
+
+       net.core.netdev_max_backlog = 182757
+
+   Apply the above sysctl.conf change via:
```
```suggestion
   Apply the above `sysctl.conf` change via:
```

ru-fu commented on doc/howto/network_increase_bandwidth.md:
```diff
@@ -46,3 +69,6 @@ To do this, use one of the following methods:
 - Set the `queue.tx.length` device option on the instance profile or configuration.
+  For example, to do this for the LXD default profile:
+
+       sudo lxc profile device set default eth0 queue.tx.length "10000" 
```
You shouldn't need sudo for this command? (If your user is in the `lxd` group, which we assume for most of the docs.)

JohnHammell replied:
Thanks for letting me know, I didn't realize the docs assumed the user was part of lxd group. Will keep this in mind for any other doc edit pull requests.

ru-fu commented on doc/howto/network_increase_bandwidth.md:
```diff
@@ -17,7 +17,30 @@ In general, you should use small `txqueuelen` values with slow devices with a hi
+
+       sudo udevadm trigger
+
+2. Increase the receive queue length (`net.core.netdev_max_backlog`).
```
```suggestion
1. Increase the receive queue length (`net.core.netdev_max_backlog`).
```
(It will show up as 2. - but using "1." everywhere is cleaner because it will update if you include a step.)

ru-fu reviewed (commented):
Thanks! This looks good - I added a few suggestions (mainly to make the spell checker happy ;) ).

I think this would work better using tabs for the different Ubuntu versions, instead of a section for each. Do you want to give this a try? But I also don't mind getting it in as is and changing it later.

JohnHammell commented:
> Thanks! This looks good - I added a few suggestions (mainly to make the spell checker happy ;) ).

Thanks for all the suggestions, I added them in.
 
> I think this would work better using tabs for the different Ubuntu versions, instead of a section for each. Do you want to give this a try? But I also don't mind getting it in as is and changing it later.

I gave it a try and think I may have it correct, but not certain as the code editor preview doesn't seem to be able to display tabbed sections. Also couldn't seem to find anything about this type of tab in the github docs. What I ended up doing was just found another page that used tabs & replicated how that was done. Hopefully I got it right.

ru-fu reviewed (commented):
Thanks a lot, this looks great! Sorry for a typo, that was my fault.

You can look at the output here: 
https://canonical-ubuntu-documentation-library--13457.com.readthedocs.build/lxd/en/13457/howto/network_increase_bandwidth/#increase-the-network-bandwidth-on-the-lxd-host

There are some more linting problems (see the doc check). If you want to run the linter locally, you can use `make doc-lint`

JohnHammell commented:
@ru-fu That link for reviewing the output, where did you get that from? I didn't see that anywhere for previewing changes. Would be good to know for future edits that I do.

Also, for the `make doc-lint` , I would like to try that, but can you point me in the right direction? I'm a bit rusty with git and submitted this pull request via the website, through the edit this document link on the page we are editing.

I setup git, gh & all that. But still not sure where to go to do the `make doc-lint` , I can `make doc` but not `make doc-lint` in the repo.

If there's a how-to page describing how to do all these things, please post a link to that and I'll see if I can figure things out better from that.

ru-fu commented:
@JohnHammell You can get to the doc preview if you look at the checks at the bottom of the PR and click the "Details" link for the Read the Docs build:
![image](https://github.com/canonical/lxd/assets/11227796/5227c9f2-4c29-483f-86d4-de0b5446ff18)

It's quite hidden, but very convenient once you know about it. :)

See here for a short intro on how to work with the documentation: https://documentation.ubuntu.com/lxd/en/latest/contributing/#contribute-to-the-documentation

If you are working locally, `make doc-lint` should run in the same way as `make doc`. What kind of error are you getting?
But it's also totally fine to use the web UI and have the check run on the PR. It might just take a few more iterations sometimes. :)

ru-fu commented on doc/howto/network_increase_bandwidth.md:
```diff
@@ -46,3 +75,6 @@ To do this, use one of the following methods:
 - Set the `queue.tx.length` device option on the instance profile or configuration.
+  For example, to do this for the LXD default profile:
+
+       lxc profile device set default eth0 queue.tx.length "10000" 
```
```suggestion
       lxc profile device set default eth0 queue.tx.length "10000"
```

ru-fu commented on doc/howto/network_increase_bandwidth.md:
```diff
@@ -18,6 +18,32 @@ For the `net.core.netdev_max_backlog` value, a good guideline is to use the mini
+
+       KERNEL=="enp5s0f1", RUN+="/sbin/ip link set %k txqueuelen 10000"
+       KERNEL=="lxdbr0", RUN+="/sbin/ip link set %k txqueuelen 10000"
+      
```
```suggestion

```

ru-fu commented on doc/howto/network_increase_bandwidth.md:
`````diff
@@ -18,6 +18,32 @@ For the `net.core.netdev_max_backlog` value, a good guideline is to use the mini
 ```
 
 ## Increase the network bandwidth on the LXD host
+````{tabs}
`````
```suggestion

````{tabs}
```

JohnHammell commented:
@ru-fu That doc preview link is definitely hard to find, but glad that I know where to find it now. It's quite useful for document change previews and probably should be more visible. Maybe even have that in place of the preview area with the code editor as that preview wasn't very good/accurate.

I'll get back to the git cli stuff later when I have a chance, for the remainder of this PR I added in those other changes you suggested via the web UI. So I think we are probably all set for now.

Let me know if there's anything else we need to do.

Thanks for all your help and giving me the chance to learn a bit more github stuff while working on this.

ru-fu reviewed (approved):
Thank you, this looks good now!

tomponline reviewed (approved):
Thanks!
